### PR TITLE
Pass Reference repo to get_source.sh

### PR DESCRIPTION
**Pull request by AdamBrousseau**
- Speed up get_source clones by using a local
  reference repository cache.

Issue #6744
[skip ci]

Signed-off-by: Adam Brousseau <adam.brousseau88@gmail.com>

### CODE REVIEW

**Comment by AdamBrousseau** [comment]
Jenkins compile plinux jdk8 depends ibmruntimes/openj9-openjdk-jdk8#415

**Comment by AdamBrousseau** [comment]
Jenkins compile plinux jdk8 depends ibmruntimes/openj9-openjdk-jdk8#415

**Comment by AdamBrousseau** [comment]
Jenkins compile plinux jdk8 depends ibmruntimes/openj9-openjdk-jdk8#415

**Comment by AdamBrousseau** [comment]
Jenkins compile plinux jdk11 depends ibmruntimes/openj9-openjdk-jdk11#315

**Comment by AdamBrousseau** [comment]
Jenkins compile plinux jdk15 depends ibmruntimes/openj9-openjdk-jdk15#2

**Comment by AdamBrousseau** [comment]
Jenkins compile plinux jdknext depends ibmruntimes/openj9-openjdk-jdk#212

**Comment by AdamBrousseau** [comment]
[Build Status](https://ci.eclipse.org/openj9/job/Build_JDK8_x86-64_linux_Personal/851/) 8
[Build Status](https://ci.eclipse.org/openj9/job/Build_JDK11_x86-64_linux_Personal/866/) 11
[Build Status](https://ci.eclipse.org/openj9/job/Build_JDK15_x86-64_linux_Personal/3/) 15
[Build Status](https://ci.eclipse.org/openj9/job/Build_JDKnext_x86-64_linux_Personal/44/) next

File: buildenv/jenkins/common/build.groovy [hunk 1]
```diff
@@ -29,9 +29,11 @@ def get_source() {
         OPENJ9_REPO_OPTION = (OPENJ9_REPO != "") ? "-openj9-repo=${OPENJ9_REPO}" : ""
         OPENJ9_BRANCH_OPTION = (OPENJ9_BRANCH != "") ? "-openj9-branch=${OPENJ9_BRANCH}" : ""
         OPENJ9_SHA_OPTION = (OPENJ9_SHA != "") ? "-openj9-sha=${OPENJ9_SHA}" : ""
+        OPENJ9_REFERENCE = "-openj9-reference=${OPENJDK_REFERENCE_REPO}"
```
Comment (jdekonin) [inline buildenv/jenkins/common/build.groovy:32]: Should this not be OPENJ9_REFERENCE?

File: buildenv/jenkins/common/build.groovy [hunk 2]
```diff
@@ -29,9 +29,11 @@ def get_source() {
...
         OMR_REPO_OPTION = (OMR_REPO != "") ? "-omr-repo=${OMR_REPO}" : ""
         OMR_BRANCH_OPTION = (OMR_BRANCH != "")? "-omr-branch=${OMR_BRANCH}" : ""
         OMR_SHA_OPTION = (OMR_SHA != "") ? "-omr-sha=${OMR_SHA}" : ""
+        OMR_REFERENCE = "-omr-reference=${OPENJDK_REFERENCE_REPO}"
```
Comment (jdekonin) [inline buildenv/jenkins/common/build.groovy:36]: Should this not be OMR_REFERENCE?

File: buildenv/jenkins/common/build.groovy (hunk 1 above)
Comment (AdamBrousseau) [reply buildenv/jenkins/common/build.groovy:32]: No, it's a single folder that holds all the repos together. Originally only had the ext repos, hence the name.

**Comment by pshipton** [comment]
The jdk14 change is missing, and jdknext needs a small update. I'll revert this, I should have merged it last anyway.

**Comment by AdamBrousseau** [comment]
I wasn't going to bother with the change for 14 but I guess I have to or else the script will puke on the unknown option. Will get that together and fix jdknext.

**Pull request merged**